### PR TITLE
fix: target cpu for torch.load in integration tests

### PR DESCRIPTION
**Pull request by sidneyw**
If models are trained on a GPU but loaded on a CPU pytorch throws an
error telling the user to explicitly target the CPU. This commit adds
the ability to pass the `map_location` parameter to `torch.load` in
order to target the CPU.

# Test Plan
- UI change:
  - [ ] add screenshots
  - [ ] React? build & check storybooks
- [ ] user-facing api change: modify documentation and examples
- [ ] user-facing api change: add the "User-facing API Change" label
- [ ] bug fix: add regression test
- [ ] bug fix: determine if there are other similar bugs in the codebase
- [ ] new feature: add test coverage for any user-facing aspects
- [ ] refactor: maintain existing code coverage

### CODE REVIEW

File: common/determined_common/experimental/checkpoint/_checkpoint.py [hunk 1]
```diff
@@ -136,12 +138,15 @@ def load(self, path: Optional[str] = None, tags: Optional[List[str]] = None) ->
                 the tensoflow saved_model. See documentation for
                 `tf.compat.v1.saved_model.load_v2
                 <https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/saved_model/load_v2>`_.
+            kwargs: Only relevant for pytorch checkpoints. The keword arguments
```
Comment (brain-good) [inline common/determined_common/experimental/checkpoint/_checkpoint.py:141]: `pytorch` -> `PyTorch`
`keword` -> `keyword`
